lib/style/module_directives.ex

```diff
@@ -65,18 +65,62 @@ defmodule Styler.Style.ModuleDirectives do
   """
   @behaviour Styler.Style
 
-  @directives ~w(alias import require)a
+  alias Styler.Zipper
+
+  @directives ~w(alias import require use)a
   @attr_directives ~w(moduledoc shortdoc behaviour)a
 
   # module names ending with these suffixes will not have a default moduledoc appended
   @dont_moduledoc ~w(Test Mixfile MixProject Controller Endpoint Repo Router Socket View HTML JSON)
   @moduledoc_false {:@, [], [{:moduledoc, [], [{:__block__, [], [false]}]}]}
 
-  def run({{:defmodule, def_meta, [name, [{mod_do, {:__block__, children_meta, children}}]]}, zipper_meta}, ctx) do
-    {directives, other} =
-      Enum.split_with(children, fn
+  def run({{:defmodule, _, children}, _} = zipper, ctx) do
+    [{_, _, aliases} = _module_name, [{mod_do, _module_body__move_focus_here!}]] = children
+    # Move the zipper's focus to the module's body
+    name = aliases |> List.last() |> to_string()
+    add_moduledoc? = not String.ends_with?(name, @dont_moduledoc)
+    body_zipper = zipper |> Zipper.down() |> Zipper.right() |> Zipper.down() |> Zipper.down() |> Zipper.right()
+
+    case Zipper.node(body_zipper) do
+      {:__block__, _, _} ->
+        {:skip, organize_directives(body_zipper, add_moduledoc?), ctx}
+
+      {:@, _, [{:moduledoc, _, _}]} ->
+        # a module whose only child is a moduledoc. nothing to do here!
+        # seems weird at first blush but lots of projects/libraries do this with their root namespace module
+        {:skip, zipper, ctx}
+
+      only_child ->
+        # There's only one child, and it's not a moduledoc. We'll maybe add a doc, then continue running this style
+        # on the `only_child` node
+        {_, do_meta, _} = mod_do
+
+        if add_moduledoc? and do_meta[:format] != :keyword do
```

Comment:
great point! i knew in the other branch that it _wasnt_ a keyword since it had multiple children, but we don't lose anything but moving it up there to be even more explicit. much good.